src/utilities/createRouter.ts

```diff
@@ -38,67 +35,60 @@ export function createRouter<const T extends Routes>(routes: T, options: RouterO
 
   async function update(url: string, { replace }: RouterUpdateOptions = {}): Promise<void> {
     const to = getResolvedRouteForUrl(routerRoutes, url) ?? getRejectionRoute('NotFound')
-    const from = isRejectionRoute(route) ? null : route
-
-    try {
-      await executeRouteHooks({
-        hooks: [
-          ...hooks.before,
-          ...getRouteHooks(to, from, 'before'),
-        ],
-        to,
-        from,
-        onRouteHookError,
-      })
-
-      history.update(url, { replace })
-
-      if (isRejectionRoute(to)) {
-        reject('NotFound')
-      } else {
-        clearRejection()
-      }
-
-      updateRoute(to)
-
-      // if (!shouldRunOnAfterLocationUpdate) {
-      //   return
-      // }
-
-      // await executeRouteHooks({
-      //   hooks: [
-      //     ...hooks.after,
-      //     ...getRouteHooks(to, from, 'after'),
-      //   ],
-      //   to,
-      //   from,
-      //   onRouteHookError,
-      // })
-
-    } catch (error) {
-      if (error instanceof NavigationAbortError) {
+    const from = route
+
+    const beforeResponse = await runBeforeRouteHooks({ to, from, hooks })
+
+    const setHistory = (): void => history.update(url, { replace })
+    const setRoute = (): void => updateRoute(to)
```

Comment:
~one thing I noticed with this change is now that we're not using an exception, the abort doesn't actually interrupt execution of other hooks. Not sure that really matters?~

never mind, looks like inside of hooks it still uses abort error, which WOULD in fact interrupt other hooks?

Reply:
I personally hate the syntax of a `const` function with the type defined inline like this. It's 1000% a nit, but I would have written it as 

```suggestion
    function setHistory(): void {
      history.update(url, { replace })
    }

    function setRoute(): void {
      updateRoute(to)
    }
```

Or possibly not even defined these, since calling the underlying functions is pretty straight forward already

Reply:
Yeah I don't feel strongly here. I was mostly defining these so that the switch had zero logic. Just called `setThing() and setOtherThing()` reads nicely. 

Logic wise it's the same either way. 

> never mind, looks like inside of hooks it still uses abort error, which WOULD in fact interrupt other hooks?

Correct. We're still throwing and catching exceptions, that just is an implementation detail of running hooks and not something you know about inside or outside of hooks.